packages/firestore/test/lite/integration.test.ts

```diff
@@ -1219,4 +1232,506 @@ describe('withConverter() support', () => {
+            {
+              // @ts-expect-error
+              nested: 3
+            },
+            { merge: true }
+          );
+        });
+      });
+
+      it('checks for nonexistent properties', async () => {
+        return withTestDb(async db => {
+          const ref = doc(collection(db, 'testobj')).withConverter(
+            testConverterMerge
+          );
+          // Top-level property.
+          await setDoc(
+            ref,
+            {
+              // @ts-expect-error
+              nonexistent: 'foo'
+            },
+            { merge: true }
+          );
+
+          // Nested property
+          await setDoc(
+            ref,
+            {
+              nested: {
+                // @ts-expect-error
+                nonexistent: 'foo'
+              }
+            },
+            { merge: true }
+          );
+        });
+      });
+    });
+
+    describe('WithFieldValue', () => {
+      it('supports FieldValues', async () => {
+        return withTestDb(async db => {
+          const ref = doc(collection(db, 'testobj')).withConverter(
+            testConverter
+          );
+
+          // Allow Field Values and nested partials.
+          await setDoc(ref, {
+            outerString: 'foo',
+            outerArr: [],
+            nested: {
+              innerNested: {
+                innerNestedNum: increment(1)
+              },
+              innerArr: arrayUnion(2),
+              timestamp: serverTimestamp()
+            }
+          });
+        });
+      });
+
+      it('requires all fields to be present', async () => {
+        return withTestDb(async db => {
+          const ref = doc(collection(db, 'testobj')).withConverter(
+            testConverter
+          );
+
+          // Allow Field Values and nested partials.
+          // @ts-expect-error
+          await setDoc(ref, {
+            outerArr: [],
+            nested: {
+              innerNested: {
+                innerNestedNum: increment(1)
+              },
+              innerArr: arrayUnion(2),
+              timestamp: serverTimestamp()
+            }
+          });
+        });
+      });
+
+      it('validates inner and outer fields', async () => {
+        return withTestDb(async db => {
+          const ref = doc(collection(db, 'testobj')).withConverter(
+            testConverter
+          );
+
+          await setDoc(ref, {
+            outerString: 'foo',
+            // @ts-expect-error
+            outerArr: 2,
+            nested: {
+              innerNested: {
+                // @ts-expect-error
+                innerNestedNum: 'string'
+              },
+              innerArr: arrayUnion(2),
+              timestamp: serverTimestamp()
+            }
+          });
+        });
+      });
+
+      it('checks for nonexistent properties', async () => {
+        return withTestDb(async db => {
+          const ref = doc(collection(db, 'testobj')).withConverter(
+            testConverter
+          );
```

Comment:
You can use `withTestDoc` here as well (and all over).

Reply:
Done.